src/Analyzers/Core/Analyzers/FileHeaders/AbstractFileHeaderDiagnosticAnalyzer.cs

```diff
@@ -92,5 +105,50 @@ private static bool CompareCopyrightText(string expectedFileHeader, string copyr
 
             return true;
         }
+
+        /// <summary>
+        /// Make sure that both \n and \r\n are accepted from the settings and leading and trailing whitespace is ignored.
+        /// </summary>
+        /// <param name="original">The original header text from the document or editorconfig</param>
+        /// <returns>A normalized text for comparison.</returns>
+        private static string NormalizeCopyrightText(string original)
+            => original.Trim().Replace("\r\n", "\n");
+
+        private FileHeader ParseFileHeader(SyntaxNode root)
+        {
+            var banner = SyntaxFacts.GetFileBanner(root);
+            if (banner.Length == 0)
+            {
+                return GetMissingHeader(root);
+            }
+
+            using var _ = PooledStringBuilder.GetInstance(out var sb);
+            int? fileHeaderStart = null;
+            int? fileHeaderEnd = null;
+            string? commentPrefix = null;
+
+            foreach (var trivia in banner)
+            {
+                if (SyntaxFacts.IsRegularComment(trivia))
+                {
+                    var comment = SyntaxFacts.GetCommentText(trivia);
+                    fileHeaderStart ??= trivia.FullSpan.Start;
+                    fileHeaderEnd = trivia.FullSpan.End;
+                    commentPrefix ??= SyntaxFacts.GetCommentPrefix(trivia);
+
+                    sb.AppendLine(comment.Trim());
+                }
+            }
+
+            return fileHeaderStart is int start && fileHeaderEnd is int end && commentPrefix is string { Length: var commentPrefixLength }
+                ? new FileHeader(sb.ToString(), start, end, commentPrefixLength)
+                : GetMissingHeader(root);
```

Comment:
feels like we can unify the GetMissingHeader case from above with this.

Reply:
Refactored in 381d350c